sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -526,6 +533,20 @@ abstract class CastBase extends UnaryExpression with TimeZoneAwareExpression wit
       buildCast[UTF8String](_, s => IntervalUtils.safeStringToInterval(s))
   }
 
+  private[this] def castToDayTimeInterval(from: DataType): Any => Any = from match {
+    case x: IntegralType if ansiEnabled =>
```

Comment:
We don't need to check the ANSI flag. The ANSI intervals suppose the ANSI mode. Just use exact ops.

Reply:
> We don't need to check the ANSI flag. The ANSI intervals suppose the ANSI mode. Just use exact ops. So, we don't have to support non-ANSI mode for ANSI intervals.

So only convert to these type we don't need check ANSI enabled, other convert also need to check?

Reply:
The same in other places. Just remove the code where you don't check the flag, leave the cases when you check the flag but remove flag checking. So, for the ANSI intervals, we should be always in the strict mode, and prevent silent overflows.